packages/test/test-end-to-end-tests/src/test/summarizerWithLocalChanges.spec.ts

```diff
@@ -0,0 +1,181 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { strict as assert } from "assert";
+import { DataObject, DataObjectFactory } from "@fluidframework/aqueduct";
+import { IContainer } from "@fluidframework/container-definitions";
+import { IContainerRuntimeOptions, ISummarizer } from "@fluidframework/container-runtime";
+import { IRequest } from "@fluidframework/core-interfaces";
+import { SharedMatrix } from "@fluidframework/matrix";
+import { SharedMap } from "@fluidframework/map";
+import {
+	ITestObjectProvider,
+	waitForContainerConnection,
+	summarizeNow,
+	createSummarizerFromFactory,
+} from "@fluidframework/test-utils";
+import {
+	describeNoCompat,
+	getContainerRuntimeApi,
+	itExpects,
+} from "@fluid-internal/test-version-utils";
+import { IContainerRuntimeBase, IFluidDataStoreFactory } from "@fluidframework/runtime-definitions";
+import { requestFluidObject } from "@fluidframework/runtime-utils";
+import { FluidDataStoreRuntime, mixinSummaryHandler } from "@fluidframework/datastore";
+import { pkgVersion } from "../packageVersion";
+
+const runtimeOptions: IContainerRuntimeOptions = {
+	summaryOptions: {
+		summaryConfigOverrides: { state: "disabled" },
+	},
+	gcOptions: { gcAllowed: true },
+};
+export const rootDataObjectType = "@fluid-example/rootDataObject";
+export const TestDataObjectType1 = "@fluid-example/test-dataStore1";
+export const TestDataObjectType2 = "@fluid-example/test-dataStore2";
+class TestDataObject2 extends DataObject {
+	public get _root() {
+		return this.root;
+	}
+	public get _context() {
+		return this.context;
+	}
+	// If this datastore created DDSes it would be even more nodes created without running gc
```

Comment:
>// If this datastore created DDSes it would be even more nodes created without running gc [](http://example.com/codeflow?start=1&length=89)

? do we need this comment ?

Reply:
Probably not